File: Swift2/Sources/Swift2/BitValueSequencer.swift

public class CBitValueSequencer : CValueSequencer {

	init(nValue: Int) {
		super.init(nInnerSequencers: 0, nValues: 2, nValue: nValue)
		m_nSequences = Sequences.nBitSequences;
		m_nRatios = Sequences.nBitRatios;
		Value = nValue
	}

	public override func UpdateOuterValues() {
		m_pvsParent.UpdateOuterValues()
	}
}

File: ValueSequencer.py/bit_value_sequencer.py
from __future__ import annotations

from .value_sequencer import CValueSequencer
from .sequences import Sequences


class CBitValueSequencer(CValueSequencer):
    def __init__(self, value: int) -> None:
        super().__init__(0, 2, value)
        self._nSequences = Sequences.nBitSequences
        self._nRatios = Sequences.nBitRatios

    def update_outer_values(self) -> None:
        if self._pvsParent:
            self._pvsParent.update_outer_values()
